docs/src/main/sphinx/admin/fault-tolerant-execution.rst

```diff
@@ -0,0 +1,338 @@
+========================
+Fault-tolerant execution
+========================
+
+By default, if a Trino node lacks the resources to execute on a task or
+otherwise fails during query execution, the query fails and must be run again
+manually. The longer the runtime of a query, the more likely it is to be
+susceptible to such failures.
+
+Fault-tolerant execution is a mechanism in Trino that enables a cluster to
+mitigate query failures by retrying queries or their component tasks in
+the event of failure. With fault-tolerant execution enabled, in-progress query
+data is spooled and can be re-used by another worker in the event of a worker
+outage or other fault during query execution.
+
+.. note::
+
+    Fault tolerance does not apply to user error. For example, a query that
+    fails because its SQL cannot be parsed is not fault-tolerant.
+
+Configuration
+-------------
+
+Fault-tolerant execution is disabled by default. To enable the feature, set the
+``retry-policy`` configuration property to either ``QUERY`` or ``TASK``
+depending on the desired :ref:`retry policy <fte-retry-policy>`.
+
+.. code-block:: properties
+
+    retry-policy=QUERY
+
+The following configuration properties control the behavior of fault-tolerant
+execution on a Trino cluster:
+
+.. list-table:: Fault-tolerant execution configuration properties
+   :widths: 30, 40, 30
+   :header-rows: 1
+
+   * - Property name
+     - Description
+     - Default value
+   * - ``retry-policy``
+     - Configures what is retried in the event of failure, either
+       ``QUERY`` to retry the whole query, or ``TASK`` to retry tasks
+       individually if they fail. See :ref:`retry policy <fte-retry-policy>` for
+       more information.
+     - ``NONE``
+   * - ``exchange.deduplication-buffer-size``
+     - Size of the buffer for data sent from worker nodes to the coordinator. If
+       this buffer is exceeded, the query fails. This buffer is not used if an
```

Comment:
This is not accurate. We still use the buffer even if exchange is configured. But only with exchange we can handle the case when we run out of space in the buffer and fall-back to spilling data into persistent storage.

Reply:
Fixed. Is it true that a query fails if the it causes the buffer to be exceeded, without an exchange manager fall-back configured?

Reply:
yes